### PR TITLE
Add the hairbrush and allow player to groom self

### PR DESCRIPTION
**Pull request by mount2010**
#### Summary
SUMMARY: Content "Adds the hairbrush and allow players to groom themselves."

#### Purpose of change
Allow the player to groom themselves.

#### Describe the solution

I added the hairbrush item and the grooming mechanic. The player can activate the hairbrush to groom themselves, giving them the groomed effect for 8 hours. The player can lengthen this effect by wearing hairpins to keep their hair tidy. Players with the stylish effect get additional bonuses.

#### Additional context

The grooming mechanic at current does not account for different "hair" from mutations like spines or feathers. It does however account for the bald trait. This is because I cannot find an easy way to get all hair mutations.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -3489,5 +3520,4 @@ void activity_handlers::tree_communion_do_turn( player_activity *act, player *p
         q.pop();
     }
     p->add_msg_if_player( m_info, _( "The trees have shown you what they will." ) );
-    act->set_to_null();
```
Comment (ifreund) [inline src/activity_handlers.cpp]: Unrelated change?

**Comment by mount2010** [comment]
Tests failing is because someone merged a broken commit to master. Have to wait for it to be resolved

**Comment by mount2010** [comment]
I should probably make the item spawn. I forgot about that.

**Comment by codemime** [comment]
I'm not necessarily opposed to this change, but IMO currently it would add more tedium, than fun. Once you've found a hairbrush, it becomes a daily routine: don't forget to collect free morale points today. Hairbrushes don't consume any resources (apart from keypresses), don't produce noise which attracts enemies (like music instruments do), offer no challenge to the player.

**Comment by mount2010** [comment]
The morale bonus is quite low without Stylish. I think this would incentivise choosing Stylish -- perhaps hairpins (or hair gel) could be required.

**Comment by mount2010** [comment]
A friend of mine proposed that it could make your scalp bleed if you used it too much, and maybe make your hair matted if you don't do it often enough (of course, this would make hairbrushes neccessary...)

**Comment by Yunatan** [comment]
That's one step way from showering. Do we really need a hygiene management system?

**Comment by EirenexTheDragon** [comment]
I suggest it to be a optional thing that gives a low effect after being used constantly, so basically no effect after using it once, but a constant effect after you done it for many days in a row.  but forgetting it ruins the small morale bonus and requires the effort to be repeated. it should totally require more time for furred mutations too

**Comment by Fris0uman** [comment]
There could be a negative trait that requires you to be groomed ?

**Comment by IriTheEerie** [comment]
Or perhaps a mod? That way, those who enjoy the concept can add it, and those who do not can leave it out.

**Comment by Rail-Runner** [comment]
Maybe it'd be best add a query which shows up after player wakes up, asking whether they should automatically do all the hygiene stuff they can, with the tools available to them? For this to work, hygiene-related morale will probably need to be changed to last until next sleep would likely have happened (I propose 1 day, or some similar length), but at that point pretty much any hygiene procedures may be added without making it tedious.

**Comment by mount2010** [comment]
I'm honestly not proficent enough to design a completely new hygiene system - and thought that "hey, a hairbrush would be a cool idea". I think that at current my hairbrush could be nerfed (I do like @EirenexTheDragon's "after several days of grooming" idea, and it is not difficult to implement).

File: data/json/items/tools.json [hunk 2]
```diff
@@ -8708,6 +8708,20 @@
     "qualities": [ [ "CUT", 1 ], [ "CUT_FINE", 1 ], [ "BUTCHER", 12 ] ],
     "flags": [ "STAB", "BELT_CLIP" ]
   },
+  {
+    "id": "hairbrush",
+    "type": "TOOL",
+    "name": "hairbrush",
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: Add a `"plural_name": "hairbrushes",`.

File: data/json/morale_types.json [hunk 3]
```diff
@@ -284,6 +284,11 @@
     "type": "morale_type",
     "text": "Found kitten <3"
   },
+  {
+    "id": "morale_groomed",
+    "type": "morale_type",
+    "text": "Groomed Self"
```
Comment (Night-Pryanik) [inline data/json/morale_types.json]:
```suggestion
    "text": "Groomed self"
```

**Comment by kevingranade** [comment]
Strongly against, I'm not interested in having any kind of ongoing hygiene system until we have it managed via a system like what @Rail-Runner mentioned.
